src/js/components/Diagram/doc.js

```diff
@@ -18,6 +18,16 @@ export const doc = Diagram => {
     connections: PropTypes.arrayOf(
       PropTypes.shape({
         anchor: PropTypes.oneOf(['center', 'vertical', 'horizontal']),
+        animation: PropTypes.oneOfType([
```

Comment:
I think I am missing a few details on the intent/scope of this PR, so I am going to ask a handful of clarifying questions. We may want to schedule a call if it is easier to talk through these.

Based on the the issue description, I was expecting animation settings to be available in two forms:
- Specified at the Diagram component level (assuming the prop values would apply to all connections)
- Specified at the each connection level (my assumption is that anything specified at this level would take precedence over the component level for that connection)

Is this expectation in line with the intent of this PR?

My subsequent comments may not be relevant if my expectation is off base.

Reply:
Resolving now after our discussion and PR content update to help clarify.